File: codegen/codegen-erlang/src/test/resources/ir/aws_json_decode_get_user_response.expected.erl
%% @doc Decode AWS JSON response for smithy.beam.test.awsjson11#GetUser.
-spec decode_get_user_response(#http_response{}) -> {'ok', get_user_output()} | {'error', term()}.
decode_get_user_response(#http_response{status = 200, body = Body}) ->
    Decoded =
        case Body of
            <<>> ->
                #{};
            _ ->
                case jsone:try_decode(Body) of
                    {ok, Val, _} -> Val;
                    {error, _} -> #{}
                end
        end,
    {ok, #get_user_output{
        user_name = maps:get(<<"userName">>, Decoded, undefined)
    }};
decode_get_user_response(#http_response{status = Status, headers = RespHeaders, body = Body}) ->
    decode_get_user_response_error(Status, RespHeaders, Body).

File: codegen/codegen-elixir/src/test/resources/ir/aws_json_decode_get_user_response.expected.ex
@doc "Decode AWS JSON response for smithy.beam.test.awsjson11#GetUser."
@spec decode_get_user_response(map()) ::
        {:ok, Json11serviceTypes.GetUserOutput.t()} | {:error, term()}
def decode_get_user_response(%RuntimeTypes.HttpResponse{
  status: 200,
  body: body
}) do
  decoded = if(body == "" or Kernel.is_nil(body), do: %{}, else: Jason.decode!(body))
  {:ok, %Types.GetUserOutput{user_name: Map.get(decoded, "userName")}}
end
def decode_get_user_response(%RuntimeTypes.HttpResponse{status: status, headers: headers, body: body}), do: decode_get_user_response_error(status, headers, body)